containers/src/Kokkos_DynRankView.hpp

```diff
@@ -485,9 +490,16 @@ class DynRankView : private View<DataType*******, Properties...> {
   // Types below - at least the HostMirror requires the value_type, NOT the rank
   // 7 data_type of the traits
 
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
   /** \brief  Compatible view of array of scalar types */
-  using array_type = DynRankView<
-      typename drvtraits::scalar_array_type, typename drvtraits::array_layout,
+  using array_type KOKKOS_DEPRECATED_WITH_COMMENT("Use type instead.") =
+      DynRankView<
+          typename drvtraits::data_type, typename drvtraits::array_layout,
+          typename drvtraits::device_type, typename drvtraits::memory_traits>;
+#endif
```

Comment:
Move this definition below and do
```C++
 using array_type KOKKOS_DEPRECATED_WITH_COMMENT("Use type instead.") = type;
```
which more clearly established they are the same as compared to repeating the definition and excepting the reader to compare them.

Reply:
Updated.